webapp/platform/components/src/compass/components/button/button.overrides.ts

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {alpha, ComponentsProps, ComponentsVariants, emphasize, Theme} from '@mui/material';
+import {ComponentsOverrides} from '@mui/material/styles/overrides';
+
+import {blend} from '../../utils/color';
+
+const componentName = 'MuiButton';
+
+declare module '@mui/material/Button' {
+    interface ButtonPropsSizeOverrides {
+        'x-small': true;
+    }
+
+    interface ButtonPropsVariantOverrides {
+        quaternary: true;
+    }
+}
+
+const defaultProps: ComponentsProps[typeof componentName] = {
+    disableElevation: true,
+    disableRipple: true,
+    disableTouchRipple: true,
+    disableFocusRipple: true,
+};
+
+const getFocusStyles = (color: string) => ({
+    '&:not(.Mui-disabled)': {
+        '&:focus': {
+            boxShadow: `inset 0 0 0 2px ${emphasize(color, 0.05)}`,
+        },
+
+        '&:focus:not(:focus-visible)': {
+            boxShadow: 'none',
+        },
+
+        '&:focus:focus-visible': {
+            boxShadow: `inset 0 0 0 2px ${emphasize(color, 0.05)}`,
+        },
+    },
+});
+
+const styleOverrides: ComponentsOverrides<Theme>[typeof componentName] = {
```

Comment:
I think my previous comment here got lost in all the renaming (thanks for doing that by the way!):

Is this something we plan to keep in the longer term? It seems weird to me that we're having to inject some overrides into the Provider instead of doing these overrides on the component itself

Reply:
I will discuss this with Nevy, Mostly we will do if something can be done in https://github.com/mattermost/mattermost-server/pull/23238.